beacon-chain/rpc/eth/helpers/sync.go

```diff
@@ -15,19 +15,32 @@ import (
 
 // ValidateSync checks whether the node is currently syncing and returns an error if it is.
 // It also appends syncing info to gRPC headers.
-func ValidateSync(ctx context.Context, syncChecker sync.Checker, headFetcher blockchain.HeadFetcher, timeFetcher blockchain.TimeFetcher) error {
+func ValidateSync(ctx context.Context, syncChecker sync.Checker, headFetcher blockchain.HeadFetcher, timeFetcher blockchain.TimeFetcher, optimisticModeFetcher blockchain.OptimisticModeFetcher) error {
 	if !syncChecker.Syncing() {
 		return nil
 	}
 	headSlot := headFetcher.HeadSlot()
-	syncDetailsContainer := &SyncDetailsContainer{
-		SyncDetails: &SyncDetails{
+	isOptimistic := false
+
+	headState, err := headFetcher.HeadState(ctx)
```

Comment:
I'm not confident about this - if `optimisticModeFetcher` isn't properly instantiated by `services.go` for all paths, this could throw errors. I haven't yet inspected all paths yet, pausing on this until I get a spot-check on my progress.

Reply:
It's OK, we assume things are properly initialized.